Add layout attribute as metadata

paraita opened the pull request.

paraita committed:
Add task layout into the workflow metadata

paraita committed:
Updated unit test for the layout metadata

paraita committed:
Updated integration test for the layout metadata

paraita committed:
Resolving conflicts due to sync with master branch

paraita commented:
@ow2-proactive/developers can I have 2 reviews please ?

marcocast commented on src/main/java/org/ow2/proactive/workflow_catalog/rest/service/WorkflowRevisionService.java:
```diff
@@ -126,11 +126,13 @@ public WorkflowMetadata createWorkflowRevision(Long bucketId, Optional<Long> wor
 
         Workflow workflow = null;
         WorkflowRevision workflowRevision;
+        String layoutValue = layout.isPresent() ? layout.get() : "";
```
You can use the orElse functionality from the Optional monad here : 

String layoutValue = layout.orElse("")

marcocast commented:
Seems ok to me

laurentpellegrino commented on src/integration-test/java/org/ow2/proactive/workflow_catalog/rest/controller/WorkflowControllerIntegrationTest.java:
```diff
@@ -74,6 +74,10 @@
 
     private static final String WORKFLOW_RESOURCE = "/buckets/{bucketId}/workflows/{workflowId}";
 
+    private static final String LAYOUT =
```
I would have renamed all references to layout name to a new name like _layoutMetadata_ to better assess there is nothing related to a UI element associated to the workflow catalog.

paraita committed:
Simplify code and fixing typos

Use orElse() instead of conditional with Optionals and fix the layout attribute name in the test

paraita commented:
Thanks guys for your comments :+1:
